stdlib/public/runtime/Backtrace.cpp

```diff
@@ -1003,6 +1007,15 @@ _swift_spawnBacktracer(const ArgChar * const *argv)
   int ret = safe_spawn(&child, swiftBacktracePath, memserver_fd,
                        const_cast<char * const *>(argv),
                        const_cast<char * const *>(env));
+#elif defined(__FreeBSD__)
+  posix_spawn_file_actions_t fa;
+  posix_spawn_file_actions_init(&fa);
+  posix_spawn_file_actions_adddup2(&fa, memserver_fd, 4);
+  posix_spawn_file_actions_addclose(&fa, memserver_fd);
+  int ret = posix_spawn(&child, swiftBacktracePath,
+                        &fa, nullptr,
+                        const_cast<char * const *>(argv),
+                        const_cast<char * const *>(env));
```

Comment:
At least on Darwin, `posix_spawn_file_actions_xxx()` allocates memory, which we can't do here because this code is run when the program has crashed (which might happen while holding a memory allocator lock, for instance).

I'd also want to check exactly what `posix_spawn()` itself does internally on FreeBSD. AFAIK it isn't defined by POSIX as async-signal-safe; if it's a syscall, it's probably fine, but if not then we might need something else here instead.

Reply:
Yeah, unfortunately not; vfork (-ish; same semantics) + exec